### PR TITLE
[feat][broker] PIP-398: Subscription replication on the namespace and topic levels

### PR DESCRIPTION
**Pull request by nodece**
PIP: https://github.com/apache/pulsar/pull/23770

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java [hunk 1]
```diff
@@ -278,6 +279,10 @@ protected void updateTopicPolicyByNamespacePolicy(Policies namespacePolicies) {
         updateSchemaCompatibilityStrategyNamespaceValue(namespacePolicies);
         updateNamespaceDispatchRate(namespacePolicies, brokerService.getPulsar().getConfig().getClusterName());
         topicPolicies.getResourceGroupName().updateNamespaceValue(namespacePolicies.resource_group_name);
+        if (!isSystemTopic()) {
+            topicPolicies.getReplicateSubscriptionState()
```
Comment (dao-jun) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java:283]: Move to the upper of the method

**Review by dao-jun — APPROVED** [review]
Overall LGTM, just a minor comment

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 2]
```diff
@@ -5016,6 +5016,29 @@ protected void internalTruncateTopic(AsyncResponse asyncResponse, boolean author
         }
     }
 
+    protected CompletableFuture<Void> internalSetEnableReplicatedSubscription(Boolean enabled,
+                                                                              boolean isGlobal) {
```
Comment (dao-jun) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:5020]: Need to `validatePoliciesReadOnlyAccessAsync ` ?

Comment (nodece) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:5020]: This method is only used for zk, not system topic.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java (hunk 1 above)
Comment (nodece) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java]: Done.

**Review by liudezhi2098 — APPROVED** [review]

**Pull request merged**